ipykernel/kernelbase.py

```diff
@@ -806,14 +885,17 @@ async def connect_request(self, socket, ident, parent):
 
     @property
     def kernel_info(self):
-        return {
+        info = {
             "protocol_version": kernel_protocol_version,
             "implementation": self.implementation,
             "implementation_version": self.implementation_version,
             "language_info": self.language_info,
             "banner": self.banner,
             "help_links": self.help_links,
         }
+        if self._supports_kernel_subshells():
+            info["supported_features"] = ["kernel subshells"]
+        return info
```

Comment:
I need to read the enhancement proposal, but maybe always include `info["supported_features"]` even if empty list ?

Reply:
The [supported features JEP](https://github.com/jupyter/enhancement-proposals/blob/master/92-jupyter-optional-features/jupyter-optional-features.md) isn't explicit about this, so I assume we can choose to add it even if it is an empty list.

Reply:
Always return `supported_features` in 924015a.